docs/docsite/rst/porting_guides/porting_guide_core_2.19.rst

```diff
@@ -5,24 +5,546 @@
+
+.. code-block:: yaml+jinja
+
+    - assert:
+        that: inventory_hostname {{ comparison }} 'localhost'
+      vars:
+        comparison: ==
+
+
+The error reported is::
+
+    Syntax error in expression. Template delimiters are not supported in expressions: chunk after expression
+
+
+Troubleshooting Untrusted Templates
+-----------------------------------
+
+By default, untrusted templates are silently ignored.
+To troubleshoot trust issues with templates, it can be helpful to trigger a warning or error for untrusted templates.
+The environment variable ``_ANSIBLE_TEMPLAR_UNTRUSTED_TEMPLATE_BEHAVIOR`` can be used to control this behavior.
+
+Valid options are:
+
+* ``warn`` - A warning will be issued when an untrusted template is encountered.
+* ``fail`` - An error will be raised when an untrusted template is encountered.
+* ``ignore`` - Untrusted templates are silently ignored and used as-is. This is the default behavior.
+
+.. note::
+    This optional warning and failure behavior is experimental and subject to change in future versions.
+
+
+Plugin API
+==========
+
+Deprecating Values
+------------------
+
+Plugins and Python modules can tag returned values as deprecated with the new ``deprecate_value`` function from
+``ansible.module_utils.datatag``.
+A description of the deprecated feature, optional help text, and removal timeframes can be attached to the value, which
+will appear in a runtime warning if the deprecated value is referenced in an expression.
+The warning message will include information about the module/plugin that applied the deprecation tag and the
+location of the expression that accessed it.
+
+.. code-block:: python
+
+    from ansible.module_utils.datatag import deprecate_value
+
+    ...
+
+    module.exit_json(
+        color_name=deprecate_value(
+            value="blue",
+            msg="The `color_name` return value is deprecated.",
+            help_text="Use `color_code` instead.",
+        ),
+        color_code="#0000ff",
+    )
+
+
+When accessing the `color_name` from the module result, the following warning will be shown::
+
+    [DEPRECATION WARNING]: The `color_name` return value is deprecated. This feature will be removed from the 'ns.collection.paint' module in a future release.
+    Origin: /examples/use_deprecated.yml:8:14
+
+    6
+    7     - debug:
+    8         var: result.color_name
+                   ^ column 14
+
+    Use `color_code` instead.
+
+
+Applying Template Trust to Individual Values
+--------------------------------------------
+
+String values are no longer trusted to be rendered as templates by default. Strings loaded from playbooks, vars files,
+and other built-in trusted sources are usually marked trusted by default.
+Plugins that create new string instances with embedded templates must use the new ``trust_value`` function from
+``ansible.module_utils.datatag`` to tag those values as originating from a trusted source to allow the templates to be
+rendered.
```

Comment:
Should `trust_value()` have an example or further documentation, or are you waiting until the public API is more complete/stable?

Reply:
We're still working out where the actual docs for those things will live- none of the current options are great :neutral_face: